commit d72507cb400c2ef8940ee194e46818d9ebf4187f
Author: Timo Aaltonen <tjaalton@debian.org>
Date:   Mon Mar 4 11:32:22 2019 +0200

    Revert "util/u_queue: add UTIL_QUEUE_INIT_SET_FULL_THREAD_AFFINITY"
    
    This reverts commit d877451b48a59ab0f9a4210fc736f51da5851c9a.

--- a/src/gallium/drivers/radeonsi/si_pipe.c
+++ b/src/gallium/drivers/radeonsi/si_pipe.c
@@ -1019,8 +1019,7 @@ radeonsi_screen_create_impl(struct radeo
 
 	if (!util_queue_init(&sscreen->shader_compiler_queue, "sh",
 			     64, num_comp_hi_threads,
-			     UTIL_QUEUE_INIT_RESIZE_IF_FULL |
-			     UTIL_QUEUE_INIT_SET_FULL_THREAD_AFFINITY)) {
+			     UTIL_QUEUE_INIT_RESIZE_IF_FULL)) {
 		si_destroy_shader_cache(sscreen);
 		FREE(sscreen);
 		return NULL;
@@ -1030,7 +1029,6 @@ radeonsi_screen_create_impl(struct radeo
 			     "shlo",
 			     64, num_comp_lo_threads,
 			     UTIL_QUEUE_INIT_RESIZE_IF_FULL |
-			     UTIL_QUEUE_INIT_SET_FULL_THREAD_AFFINITY |
 			     UTIL_QUEUE_INIT_USE_MINIMUM_PRIORITY)) {
 	       si_destroy_shader_cache(sscreen);
 	       FREE(sscreen);
--- a/src/util/disk_cache.c
+++ b/src/util/disk_cache.c
@@ -377,8 +377,7 @@ disk_cache_create(const char *gpu_name,
     */
    util_queue_init(&cache->cache_queue, "disk$", 32, 1,
                    UTIL_QUEUE_INIT_RESIZE_IF_FULL |
-                   UTIL_QUEUE_INIT_USE_MINIMUM_PRIORITY |
-                   UTIL_QUEUE_INIT_SET_FULL_THREAD_AFFINITY);
+                   UTIL_QUEUE_INIT_USE_MINIMUM_PRIORITY);
 
    cache->path_init_failed = false;
 
--- a/src/util/u_queue.c
+++ b/src/util/u_queue.c
@@ -241,20 +241,6 @@ util_queue_thread_func(void *input)
 
    free(input);
 
-#ifdef HAVE_PTHREAD_SETAFFINITY
-   if (queue->flags & UTIL_QUEUE_INIT_SET_FULL_THREAD_AFFINITY) {
-      /* Don't inherit the thread affinity from the parent thread.
-       * Set the full mask.
-       */
-      cpu_set_t cpuset;
-      CPU_ZERO(&cpuset);
-      for (unsigned i = 0; i < CPU_SETSIZE; i++)
-         CPU_SET(i, &cpuset);
-
-      pthread_setaffinity_np(pthread_self(), sizeof(cpuset), &cpuset);
-   }
-#endif
-
    if (strlen(queue->name) > 0) {
       char name[16];
       snprintf(name, sizeof(name), "%s%i", queue->name, thread_index);
--- a/src/util/u_queue.h
+++ b/src/util/u_queue.h
@@ -48,7 +48,6 @@ extern "C" {
 
 #define UTIL_QUEUE_INIT_USE_MINIMUM_PRIORITY      (1 << 0)
 #define UTIL_QUEUE_INIT_RESIZE_IF_FULL            (1 << 1)
-#define UTIL_QUEUE_INIT_SET_FULL_THREAD_AFFINITY  (1 << 2)
 
 #if defined(__GNUC__) && defined(HAVE_LINUX_FUTEX_H)
 #define UTIL_QUEUE_FENCE_FUTEX
